envoy-control-core/src/main/java/pl/allegro/tech/servicemesh/envoycontrol/SimpleCache.java

```diff
@@ -268,12 +268,15 @@ public DeltaWatch createDeltaWatch(
                                                            String version) {
         long watchId = watchCount.incrementAndGet();
         status.setWatch(watchId, watch);
-        watch.setStop(() -> status.removeWatch(watchId));
+        watch.setStop(() -> {
+            LOGGER.debug("removing watch {}", watchId);
```

Comment:
how will this help you with investigating the issue of removing snapshots with existing watches?

Reply:
snapshot is getting cleared when watch is removed and id of watch can be connected to the event when it was created while that one can be connected to events when instances of a services were changed, not sure if that will be enough, but right now there's literally nothing that can help debug it